src/gt4py/next/iterator/type_inference.py

```diff
@@ -499,6 +546,54 @@ def visit_FunCall(
                 )
                 self.constraints.add((tup, val))
                 return Val(kind=kind, dtype=elem, size=size)
+            if node.fun.id == "make_list":
+                elems = list(self.visit(arg) for arg in node.args)
+                # TODO constraints between dtypes of args
+                lst = ValueList(
+                    dtype=elems[0].dtype,
+                    max_length=Length(length=len(elems)),
+                    has_skip_values=TypeVar.fresh(),
+                )
+                return Val(kind=Value(), dtype=lst, size=TypeVar.fresh())
+            if node.fun.id == "neighbors":
+                if len(node.args) != 2:
+                    raise TypeError("neighbors requires exactly two arguments")
+                if not isinstance(node.args[0], ir.OffsetLiteral):
+                    raise TypeError("The first argument to neighbors must be an OffsetLiteral.")
+                if not isinstance(node.args[0].value, str):
+                    raise TypeError("The first argument to neighbors must be an OffsetLiteral tag.")
+
+                max_length: Type = TypeVar.fresh()
+                has_skip_values: Type = TypeVar.fresh()
+                if self.offset_provider:
+                    connectivity = self.offset_provider[node.args[0].value]
+                    assert isinstance(connectivity, Connectivity)
+                    max_length = Length(length=connectivity.max_neighbors)
+                    has_skip_values = BoolType(value=connectivity.has_skip_values)
+                current_loc_in, current_loc_out = _infer_shift_location_types(
+                    [node.args[0]], self.offset_provider, self.constraints
+                )
+                dtype_ = TypeVar.fresh()
+                size = TypeVar.fresh()
+                it = self.visit(node.args[1])
```

Comment:
Actually it propagates the symtable. maybe there is a problem with that pass because symtable should be propagated implicitly...
Leaving this open for investigation.